bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/renault/api/MyRenaultHttpSession.java

```diff
@@ -0,0 +1,224 @@
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.util.Fields;
+import org.openhab.binding.renault.internal.RenaultConfiguration;
+import org.openhab.binding.renault.internal.renault.api.exceptions.RenaultException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+
+/**
+ * This is a Java version of the python renault-api project developed here:
+ * https://github.com/hacf-fr/renault-api
+ *
+ * @author Doug Culnane - Initial contribution
+ */
+@NonNullByDefault
+public class MyRenaultHttpSession {
+
+    private Car car = new Car();
+    private RenaultConfiguration config;
+    private HttpClient httpClient;
+    private @Nullable Constants constants;
+    private @Nullable String kamereonToken;
+    private @Nullable String kamereonaccountId;
+    private @Nullable String cookieValue;
+    private @Nullable String personId;
+    private @Nullable String gigyaDataCenter;
+    private @Nullable String jwt;
+
+    private final Logger logger = LoggerFactory.getLogger(MyRenaultHttpSession.class);
+
+    public MyRenaultHttpSession(RenaultConfiguration config, HttpClient httpClient) {
+        this.config = config;
+        this.httpClient = httpClient;
+    }
+
+    public void initSesssion() throws Exception {
+        this.constants = new Constants(config.locale);
+        httpClient.start();
+        login();
+        getAccountInfo();
+        getJWT();
+        getAccountID();
+        getVehicle();
+    }
+
+    public void updateCarData() {
+        getCockpit();
+        getBatteryStatus();
+        getLocation();
+        getHvacStatus();
+    }
+
+    private void login() throws RenaultException, InterruptedException, ExecutionException, TimeoutException {
+        Fields fields = new Fields();
+        fields.add("ApiKey", this.constants.getGigyaApiKey());
+        fields.add("loginID", config.myRenaultUsername);
+        fields.add("password", config.myRenaultPassword);
+        logger.debug("URL: {}/accounts.login", this.constants.getGigyaRootUrl());
+        ContentResponse response = httpClient.FORM(this.constants.getGigyaRootUrl() + "/accounts.login", fields);
+        if (HttpStatus.OK_200 == response.getStatus()) {
+            JsonObject responseJson = JsonParser.parseString(response.getContentAsString()).getAsJsonObject();
+            JsonObject sessionInfoJson = responseJson.getAsJsonObject("sessionInfo");
+            cookieValue = sessionInfoJson.get("cookieValue").getAsString();
```

Comment:
I assume there are possible exceptions for these 3 lines that are not listed in the method prototype ? To be added in this case.
Same remark for the other methods.

Reply:
I think I got then all now.